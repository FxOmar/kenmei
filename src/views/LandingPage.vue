<template lang="pug">
  #landing-page.font-sans.flex.flex-col.h-full
    content.flex-1
      hero-section(@signOnOpened="openSignOnWith($event)")
      features#features
      stats#stats
      call-to-action(@signOnOpened="openSignOnWith($event)")
    base-footer
    base-modal(
      :visible="signOnVisible"
      :loading="loading"
      @dialogClosed="signOnVisible = false"
      size="xs"
    )
      template(slot='body')
        component(
          :is='activeComponent'
          @componentChanged='activeComponent = $event'
          @signOnFinished='signOnVisible = false'
          @loading='loading = $event'
        )
</template>

<script>
  import HeroSection from '@/components/landing_page/LandingPageHeroSection';
  import Features from '@/components/landing_page/LandingPageFeatures';
  import Stats from '@/components/landing_page/LandingPageStats';
  import CallToAction from '@/components/landing_page/LandingPageCTA';
  import TheSignIn from '@/components/TheSignIn';
  import TheSignUp from '@/components/TheSignUp';
  import TheResetPassword from '@/components/TheResetPassword';

  export default {
    components: {
      HeroSection,
      Features,
      Stats,
      CallToAction,
      TheSignIn,
      TheSignUp,
      TheResetPassword,
    },
    data() {
      return {
        loading: false,
        signOnVisible: false,
        activeComponent: 'TheSignUp',
      };
    },
    methods: {
      openSignOnWith(comp) {
        this.activeComponent = comp;
        this.signOnVisible = true;
      },
    },
  };
</script>
