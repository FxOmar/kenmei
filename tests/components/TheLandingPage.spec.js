import { shallowMount } from '@vue/test-utils';
import LandingPage from '@/components/TheLandingPage.vue';

describe('LandingPage.vue', () => {
  it('renders the page', () => {
    const landingPage = shallowMount(LandingPage);
    expect(landingPage.element).toMatchSnapshot();
  });
});
