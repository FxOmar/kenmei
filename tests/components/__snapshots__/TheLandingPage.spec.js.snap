// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage.vue renders the page 1`] = `
<div
  class="container mx-auto w-full h-full flex flex-col justify-center items-center"
>
  <el-card-stub
    class="text-center max-w-lg mx-5 md:mx-0"
    shadow="always"
  >
    <img
      class="max-w-xs w-full inline logo"
      src="@/assets/logo_transparent.png"
    />
    <div
      class="welcome-text font-size-b font-line-height-secondary"
    >
      <p>
        Welcome to Kenmei, a new manga tracking website currently in
development.
      </p>
      <p>
        Inspired by

        <el-link-stub
          class="inline align-baseline"
          href="https://www.trackr.moe"
          type="default"
        >
          trackr.moe
        </el-link-stub>
        , the best manga tracking website,

        <el-link-stub
          class="inline align-baseline"
          href="https://blog.codeanimu.net/posts/2019/07/trackr-moe-the-future/"
          type="default"
        >
          that is now slated to be closed
        </el-link-stub>
        , Kenmei is here to replace it
and provide even more features, for all manga fans out there!
      </p>
      <p>
        If you would like to get notified, when registrations open, please
leave your email below.
      </p>
    </div>
    <div
      class="email-form"
    >
      <form
        action="https://tinyletter.com/kenmei"
        method="post"
        onsubmit="window.open('https://tinyletter.com/kenmei', 'popupwindow', 'scrollbars=yes,width=800,height=600');"
        target="popupwindow"
      >
        <el-input-stub
          autocomplete="off"
          class="my-5"
          name="email"
          placeholder="Please enter your email"
          type="text"
          validateevent="true"
          value=""
        />
        <input
          name="embed"
          type="hidden"
          value="1"
        />
        <input
          class="el-button"
          type="submit"
          value="Subscribe"
        />
      </form>
    </div>
  </el-card-stub>
</div>
`;
